// Copyright 2015 The Android Open Source Project
//
//   Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";
option optimize_for = LITE_RUNTIME;

message UVCMessage {
  required int32  msg_type = 1;
  required string msg_name = 2;
  optional MediaBufferInfo buffer_info = 3;
  optional StreamInfo stream_info      = 4;
  optional MethodParams method_params  = 5;
}

message MediaBufferInfo {
  required int32 id       = 1;
  required int32 size     = 2;
  required int32 fd       = 3;
  optional int32 handle   = 4;
  optional int64 pts      = 5;
  optional int64 data     = 6;
  optional int64 priv_data = 7;
  optional int32 seq      = 8;
}

message StreamInfo {
  required int32 width      = 1;
  required int32 height     = 2;
  required int32 vir_width  = 3;
  required int32 vir_height = 4;
  required int32 buf_size   = 5;
  optional int32 range      = 6;
}

message MethodParams {
  optional string str_p = 1;
  optional int32  i32_p = 2;
  optional int64  i64_p = 3;
  optional float  flo_p = 4;
}